megamek/src/megamek/common/ConstructionFactorWarning.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2023 - The MegaMek Team. All Rights Reserved.
+ *
+ * This file is part of MegaMek.
+ *
+ * MegaMek is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * MegaMek is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MegaMek. If not, see <http://www.gnu.org/licenses/>.
+ */
+package megamek.common;
```

Comment:
Is there a reason for putting this in `megamek.common`? It depends on `megamek.client` and is not used by anything outside that package as far as I can see. Ideally client and server depend on common, but not the other way around. The current situation is pretty bad, but best not to make it worse.

Reply:
Agreed - this kind of evolved as an attempt to consolidate the core logic for this feature into a single unit testable place.  Is it alright if I move it to `package megamek.client.ui.swing;`

Reply:
Moved to `megamek.client.ui.swing`.